src/components/ec-option-card/ec-option-card.vue

```diff
@@ -0,0 +1,199 @@
+<template>
+  <component
+    :is="componentTag"
+    v-bind="{
+      ...attrs,
+      ...componentProps,
+      'data-test': attrs['data-test'] ? `${attrs['data-test']} ec-option-card` : 'ec-option-card',
+    }"
+    class="ec-option-card"
+    :class="getOptionCardClasses()"
+    @click="setClickAction()"
+  >
+    <span
+      data-test="ec-option-card__title"
+    >
+      <ec-icon
+        v-if="optionCardIconName && optionTitle"
+        data-test="ec-option-card__title-img"
+        class="ec-option-card__title-img"
+        :class="getOptionIconClass()"
+        :name="optionCardIconName"
+        :alt="optionTitle"
+        :size="15"
+      />{{ optionTitle }}</span>
+    <p
+      v-if="optionCaption"
+      class="ec-option-card__caption"
+      :class="getOptionCaptionClass()"
+      data-test="ec-option-card__caption"
+    >{{ optionCaption }}</p>
+  </component>
+</template>
+
+<script setup lang="ts">
+import {
+  computed,
+  toRefs,
+  useAttrs,
+} from 'vue';
+
+import EcIcon from '../ec-icon';
+import type { IconName } from '../ec-icon/types';
+import { OptionCardType } from './types';
+
+const attrs = useAttrs();
+
+interface OptionCardProps {
+  isDisabled?: boolean,
+  isSubmit?: boolean,
+  optionTitle: string,
+  optionCaption?: string,
+  optionCardIconName?: IconName,
+  optionCardType?: OptionCardType,
+  to?: string,
+  href?: string,
+}
+
+const props = defineProps<OptionCardProps>();
+
+const {
+  isDisabled,
+  isSubmit,
+  optionTitle,
+  optionCaption,
+  optionCardIconName,
+  optionCardType,
+  to,
+  href,
+} = toRefs(props);
+
+const componentTag = computed(() => {
+  if (to?.value) {
+    return 'router-link';
+  } if (href?.value) {
+    return 'a';
+  }
+  return 'button';
+});
+
+const isOptionCardDisabled = computed(() => ({ disabled: isDisabled?.value }));
+
+const optionCardButtonType = computed(() => (isSubmit?.value ? 'submit' : 'button'));
+
+const optionCardNavigationType = computed(() => (to?.value ? { to: to.value } : { href: href?.value }));
+
+const componentProps = computed(() => {
+  const newProps = {
+    ...isOptionCardDisabled.value,
+    ...(componentTag.value === 'button' && { type: optionCardButtonType.value }),
+    ...optionCardNavigationType.value,
+  };
+
+  return newProps;
+});
+
+const emit = defineEmits(['click', 'submit']);
```

Comment:
could you please have a look how emits are defined in ec-input ? We have to define it using types.

Reply:
Like this? 
```
const emit = defineEmits<{(e: 'update:modelValue', value: InputFieldEvents[InputFieldEvent.UPDATE_MODEL_VALUE]): void
  (e: 'icon-click', value: InputFieldEvents[InputFieldEvent.ICON_CLICK]): void
}>();
```
so in this case it would be 

```
const emit = defineEmits<{(e: 'click'): void
  (e: 'submit', value: ButtonValueEvent): void
}>();

```
`ButtonValueEvent` I'm not sure what it could be probably a string...

Reply:
Yes, even more, I think we could remove submit.
```
const emit = defineEmits<{(e: 'click'): void}>();
```